### PR TITLE
docs: production vs non-production use

### LINKED ISSUE #607: Create BSL clarification page
<html><head></head><body><h1>Using Omni Under the Business Source License: Production vs. Non-Production Use</h1>
<p><em>Interpretive guidance from Sidero Labs </em></p>
<h2>Why this guide exists</h2>
<p>Omni's self-hosted server is distributed under the <strong>Business Source License 1.1 (BSL / BUSL)</strong>. The BSL lets anyone read, modify, build on, and <strong>run Omni for non-production use at no charge</strong>. <strong>Production</strong> use requires a commercial license — a self-hosted Omni subscription — or you can use our hosted SaaS instead.</p>
<p>Omni's license draws the production/non-production line directly (quoted below). This guide applies that line to concrete situations, so you can get a clear, good-faith answer to <em>"do I need a license for this?"</em></p>
<p>This is interpretive guidance to help you apply the license. It is not a change to the license, and your use is ultimately governed by the BSL and any agreement you have with us.</p>
<p>For reference, Omni's Additional Use Grant reads:</p>
<blockquote>
<p>You may make non-production use of the Licensed Work, including testing and evaluation of the Licensed Work itself, or personal use in a home lab environment. Use of the Licensed Work to host, run, test, or support any environment on which Customer's own development, operations, or business depends is not non-production use and requires a commercial license, regardless of how that environment is designated.</p>
</blockquote>
<p>The grant makes two things free — evaluating Omni itself, and personal home-lab use — and treats everything else as production, expressly including any environment your own development, operations, or business depends on, <em>regardless of how that environment is designated</em>. So a staging or test fleet you rely on to build or test your own software is production use even though it's labeled "staging", as your engineers/organization would be disrupted if it went down.  </p>
<h2>The short version</h2>
<p>Ask one question:</p>
<blockquote>
<p><strong>Does your organization depend on this Omni deployment, day to day, to develop, test, run, or deliver its own software or business?</strong></p>
</blockquote>
<ul>
<li><strong>Yes →</strong> it's <strong>production</strong>. A commercial license is required (or use hosted Omni) — <em>whatever the environment is labeled (production, staging, QA, dev), and regardless of node count.</em></li>
<li><strong>No →</strong> it's <strong>non-production</strong>, and free.</li>
</ul>
<p>Non-production is the narrow, well-defined zone. Your use is free if it falls into one of these:</p>
<ul>
<li><strong>You're evaluating Omni itself</strong> — trying it, learning it, running a proof of concept or bake-off to decide whether to adopt it. Free at any scale, for as long as the evaluation genuinely lasts.</li>
<li><strong>Throwaway experimentation</strong> — a developer spins up a cluster to try something and tears it down, and nothing and no one relies on it.</li>
<li><strong>Personal home-lab use</strong> — non-commercial, for yourself.</li>
</ul>
<p>Two clarifications that resolve most edge cases:</p>
<ul>
<li><strong>The label of the environment does not decide it.</strong> "Staging," "QA," "test," and "dev" are not free passes. What decides it is whether your organization <em>depends on</em> the deployment to do its own work. A 600-node staging fleet that your engineers rely on to test your product is production, even though it is called staging and even though no customer ever connects to it.</li>
<li><strong>What you're testing decides the free zone, not the word "test."</strong> Testing <em>Omni</em> is free. Using Omni as the standing platform on which you test <em>your own software</em> is production.</li>
</ul>
<h2>Non-production (free)</h2>
<p>Typical examples:</p>
<ul>
<li>Evaluating, learning, or running a proof of concept or bake-off of <strong>Omni itself</strong></li>
<li>A developer spinning up a throwaway cluster to experiment, with nothing depending on it</li>
<li>Home labs and personal, non-commercial use</li>
<li>Internal training and upskilling — your own staff learning Omni on throwaway clusters</li>
<li>Ephemeral demo sandboxes spun up and torn down for a presentation</li>
</ul>
<h2>Production (commercial license required)</h2>
<p>Typical examples:</p>
<ul>
<li>Any standing environment your organization depends on to develop, test, run, or deliver its own software or business — <strong>production, staging, QA, UAT, and persistent dev/test platforms alike</strong></li>
<li>Clusters running customer-facing or revenue-generating applications</li>
<li>Internal applications and services your organization relies on day to day</li>
<li>CI/CD, build, artifact, and delivery infrastructure your engineering org depends on</li>
<li>Persistent staging, QA, UAT, or pre-production fleets your team relies on — even when they test only your own software and no customer connects to them</li>
<li>Disaster-recovery, standby, or failover environments for any of the above</li>
<li>Using Omni to manage clusters on behalf of your own clients (managed services)</li>
<li>Delivering training, courses, or hands-on labs as a commercial service, where students or customers connect to Omni-managed environments</li>
</ul>
<h2>Example</h2>
<p>An enterprise uses Omni to manage a fleet of <strong>1,000 production nodes</strong> running its applications and <strong>600 staging nodes</strong> used to test its own software before release. No customer ever connects to the staging fleet.</p>
<p><strong>Both require a commercial license.</strong> The production fleet is obvious. The staging fleet is production use too, because it is standing infrastructure the organization depends on day to day to test its own software — the "staging" label and the absence of external users don't change that. The only Omni use here that would be free is a developer separately spinning up a throwaway cluster to evaluate a new Omni feature, or someone running Omni in a home lab.</p>
<h2>Scenario reference</h2>

Scenario | Classification | Why
-- | -- | --
Evaluating Omni / proof of concept / bake-off (any scale) | Non-production | You're assessing Omni itself, not running your own work on it
Developer spins up a throwaway cluster to experiment, nothing relies on it | Non-production | Throwaway; the organization doesn't depend on it
Home lab / personal, non-commercial use | Non-production | No organization or customer relies on it
Internal training — staff learning Omni on throwaway clusters | Non-production | Learning Omni, nothing depends on it
Ephemeral demo sandbox for a presentation | Non-production | Spun up and torn down; nothing relies on it
Persistent staging/QA/UAT fleet testing your own software (e.g., 600 nodes) | Production | Standing infrastructure the org depends on; label and lack of external users don't matter
CI/CD, build, artifact, or delivery infrastructure the eng org depends on | Production | Standing infrastructure the organization relies on day to day
Standing internal dev platform engineers rely on every day | Production | The org depends on it, even if individual nodes are recycled
Internal apps/services your organization relies on to operate | Production | Operational dependency
Customer-facing or revenue-generating applications | Production | Real users depend on it
Disaster recovery / hot or warm standby for any of the above | Production | Exists to carry production workloads
Managing clients' clusters as a service provider | Production | Commercial service delivery
Commercial training business — students connect to Omni-managed labs they paid for | Production | Omni is the substrate of a paid service customers depend on

<h2>Edge cases worth calling out</h2>
<ul>
<li><strong>"It's only staging / test, and no customer touches it."</strong> Not free. The free "testing" is testing <em>Omni</em>. A standing staging or test fleet your organization relies on to test <em>its own</em> software is production, whatever it's named and whether or not anyone outside connects.</li>
<li><strong>Throwaway vs. standing.</strong> The line is organizational dependence, not literal uptime. A developer's scratch cluster they create and destroy is free. A platform the team relies on every day is production even if individual nodes are torn down and rebuilt nightly.</li>
<li><strong>Evaluating Omni at scale.</strong> A genuine evaluation or proof of concept of Omni itself stays non-production even if it's sizable or runs for weeks — right up until you begin relying on it to run, build, or test your own software, at which point it's production.</li>
</ul>
<h2>A note on scale</h2>
<p>There is <strong>no free production tier</strong> — no node threshold, and no "small enough to be free" amount of production use. The first standing deployment your organization depends on needs a commercial license. If you'd rather not run the server yourself, hosted Omni is licensed for production out of the box.</p>
<h2>Still not sure?</h2>
<p>Tell us what you're running and we'll give you a straight answer — we'd much rather help you classify it up front than surprise you later. Reach us at <strong>sales@siderolabs.com</strong>, or see hosted Omni pricing at <strong>siderolabs.com/pricing</strong>.</p>
<hr>
<p><em>This document reflects Sidero Labs' good-faith interpretation of "production use" under the Business Source License 1.1 for Omni and may be updated. It does not modify the license or any agreement between you and Sidero Labs.</em></p></body></html>

### PR DESCRIPTION
**Pull request by Iheanacho-ai**
fixes #607

### CODE REVIEW

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+title: "Production vs. Non-Production Use Under the Business Source License"
+description: "Interpretive guidance from Sidero Labs on applying the Business Source License's production and non-production distinction to common Omni deployment patterns."
+---
+
+Omni's self-hosted server is distributed under the Business Source License 1.1 (BSL / BUSL). The license lets anyone read, modify, and build on Omni freely, and run it for non-production use at no charge. Production use requires a commercial license: either a self-hosted Omni subscription, or our hosted SaaS, which is licensed for production by default.
```
Comment (rothgar) [inline public/omni/self-hosted/production-vs-non-production.mdx]:
```suggestion
Omni is distributed under the Business Source License 1.1 (BSL / BUSL). The license lets anyone read, modify, and build on Omni freely, and run it for non-production use at no charge. Production use requires a commercial license: either a self-hosted Omni subscription, or Sidero Labs' hosted SaaS offering, which is licensed for production by default.
```

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+The line between those two cases comes from one sentence in the license, the Additional Use Grant:
```
Comment (sterlingkoch) [inline public/omni/self-hosted/production-vs-non-production.mdx]: It's actually two sentences.

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+That single sentence leaves room for judgment calls, particularly around staging, QA, and internal test environments. The rest of this guide works through those judgment calls in practical terms.
+
+## How to tell if you need a commercial license
+
+Most cases come down to a single underlying question:
+
+> Does your organization depend on this Omni deployment, day to day, to develop, test, run, or deliver its own software or business?
+
+If the answer is yes, you're in production, and a commercial license is required, regardless of what the environment is called (production, staging, QA, dev) or how many nodes it has. If the answer is no, you're in non-production, and free to run Omni without a license.
```
Comment (rothgar) [inline public/omni/self-hosted/production-vs-non-production.mdx]:
```suggestion
If the answer is yes, you're using a production instance, and a commercial license is required, regardless of what the environment is called (production, staging, QA, dev) or how many nodes it has. If the answer is no, you're using a non-production instance, and free to run Omni without a license.
```

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+---
+
+Omni's self-hosted server is distributed under the Business Source License 1.1 (BSL / BUSL). The license lets anyone read, modify, and build on Omni freely, and run it for non-production use at no charge. Production use requires a commercial license: either a self-hosted Omni subscription, or our hosted SaaS, which is licensed for production by default.
+
+The line between those two cases comes from one sentence in the license, the Additional Use Grant:
+
+> *"You may make non-production use of the Licensed Work, including for testing and evaluation, or for personal use in a home lab environment. Any use of the Licensed Work in a production environment requires a commercial license from Sidero Labs."*
+
+<Note>
+  That sentence is the entirety of what the license says about production vs. non-production use. Everything below it in this guide is Sidero Labs' interpretation of how that one sentence applies to real deployments, not additional license text.
```
Comment (sterlingkoch) [inline public/omni/self-hosted/production-vs-non-production.mdx]: It's two sentences.  Suggestion: change to "That grant is the entirety..."

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+</Note>
+
+That single sentence leaves room for judgment calls, particularly around staging, QA, and internal test environments. The rest of this guide works through those judgment calls in practical terms.
```
Comment (sterlingkoch) [inline public/omni/self-hosted/production-vs-non-production.mdx]: The license's second sentence is precisely what resolves the staging/QA question ("regardless of how that environment is designated"). With the correct text in place, this should be reframed the way Steve did in the issue. The grant makes two things free (evaluating Omni itself, and home-lab use) and treats everything else as production, expressly including any environment your business depends on regardless of label. The guide then applies that to concrete cases. The judgment calls are in mapping real deployments, not in the license being vague.

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 6]
```diff
@@ -0,0 +1,98 @@
+---
+title: "Production vs. Non-Production Use Under the Business Source License"
```
Comment (sterlingkoch) [inline public/omni/self-hosted/production-vs-non-production.mdx:2]: Style guide says sentence case, yes?

File: public/omni/self-hosted/production-vs-non-production.mdx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+| Commercial training business, students connect to Omni-managed labs they paid for | Production | Omni is the substrate of a paid service customers depend on |
+
+Even at this scale, the underlying rule stays the same: there is no node count or usage level small enough to stay free once a deployment qualifies as production, and there is no free production tier regardless of size. If you would rather not run the server yourself, hosted Omni is licensed for production use by default.
+
+If you are not sure where your deployment falls, contact us with details about what you are running and we can help classify it. Reach us at sales@siderolabs.com, or see hosted Omni pricing at [siderolabs.com/pricing](https://www.siderolabs.com/pricing).
+
+---
+
+*This document reflects Sidero Labs' good-faith interpretation of "production use" under the Business Source License 1.1 for Omni, and may be updated over time. It does not modify the license or any agreement between you and Sidero Labs. The Additional Use Grant quoted above is the authoritative license text; everything else in this guide is interpretation.*
```
Comment (sterlingkoch) [inline public/omni/self-hosted/production-vs-non-production.mdx]: Nit: add a trailing newline at end of file — GitHub flags "No newline at end of file."

**Review by sterlingkoch — CHANGES_REQUESTED** [review]

File: public/omni/self-hosted/production-vs-non-production.mdx (hunk 2 above)
Comment (sterlingkoch) [reply public/omni/self-hosted/production-vs-non-production.mdx]: Suggested (for this whole block) to fix the two-sentence thing and also to make it a direct quote of the license (which is currently misquoted):

-------------------------

The line between those two cases comes directly from the license's Additional Use Grant:

*"You may make non-production use of the Licensed Work, including testing and evaluation of the Licensed Work itself, or personal use in a home lab environment. Use of the Licensed Work to host, run, test, or support any environment on which Customer's own development, operations, or business depends is not non-production use and requires a commercial license, regardless of how that environment is designated."*

<Note>
  That grant is the entirety of what the license says about production vs. non-production use. Everything below it in this guide is Sidero Labs' interpretation of how the grant applies to real deployments, not additional license text.
</Note>

The grant makes two things free — evaluating Omni itself, and personal home-lab use — and treats everything else as production, expressly including any environment your own development, operations, or business depends on, regardless of how it's labeled. The judgment calls that remain are about applying that line to real deployments, particularly standing staging, QA, and internal test environments. The rest of this guide works through them in practical terms.

**Review by sterlingkoch — APPROVED** [review]
